internal/k8s/controllers/pool_controller.go

```diff
@@ -109,5 +117,6 @@ func (r *PoolReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&metallbv1beta1.IPAddressPool{}).
 		Watches(&source.Kind{Type: &metallbv1beta1.AddressPool{}}, &handler.EnqueueRequestForObject{}).
 		Watches(&source.Kind{Type: &metallbv1beta1.Community{}}, &handler.EnqueueRequestForObject{}).
+		Watches(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestForObject{}).
```

Comment:
We need a predicate that discard events on the namespace that do not involve changing labels.